contracts/src/libraries/token/erc677.cairo

```diff
@@ -30,37 +37,42 @@ mod ERC677 {
     #[event]
     #[derive(Drop, starknet::Event)]
     enum Event {
-        Transfer: Transfer,
+        TransferAndCall: TransferAndCall,
     }
 
     #[derive(Drop, starknet::Event)]
-    struct Transfer {
+    struct TransferAndCall {
         from: ContractAddress,
         to: ContractAddress,
         value: u256,
         data: Array<felt252>
     }
 
-    fn transfer_and_call(
-        ref self: ContractState, to: ContractAddress, value: u256, data: Array<felt252>
-    ) -> bool {
-        let sender = starknet::info::get_caller_address();
+    #[embeddable_as(ERC677Impl)]
```

Comment:
from the cairo-book:

```Note: To avoid the confusion between the embeddable name and the impl name, we recommend keeping the suffix Impl in the impl name.```

So we should switch to
`#[embeddable_as(ERC677)]`
and
`impl ERC677Impl<...`